cmd/edit/machinepool/machinepool.go

```diff
@@ -96,56 +115,32 @@ func editMachinePool(cmd *cobra.Command, machinePoolID string, clusterKey string
 		os.Exit(1)
 	}
 
-	autoscaling, replicas, minReplicas, maxReplicas := getMachinePoolReplicas(cmd, r.Reporter, machinePoolID,
-		machinePool.Replicas(), machinePool.Autoscaling())
-
-	if !autoscaling && replicas < 0 ||
-		(autoscaling && cmd.Flags().Changed("min-replicas") && minReplicas < 0) {
-		r.Reporter.Errorf("The number of machine pool replicas needs to be a non-negative integer")
-		os.Exit(1)
+	// if no value set enter interactive mode
+	if !(isMinReplicasSet || isMaxReplicasSet || isReplicasSet || isAutoscalingSet || isLabelsSet || isTaintsSet) {
```

Comment:
Should we extract this check to line `35`?
Then we don't have to check the condition for default machine pool and non-default.

Reply:
I see your point, but I want to make sure we won't miss anything there.

Let's see:
If the user doesn't set anything - enable interactive.
If the user sets taints for default they will get an error.

Yes, I think that would work.
Changing

Reply:
fixed